src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisWorkerService.java

```diff
@@ -160,14 +158,14 @@ private SensitivityAnalysisResult run(SensitivityAnalysisRunContext context, UUI
             reporter = rootReporter.createSubReporter(SENSI_TYPE_REPORT, SENSI_TYPE_REPORT + " (${providerToUse})", "providerToUse", sensitivityAnalysisRunner.getName());
         }
 
-        SensitivityAnalysisInput sensitivityAnalysisInput = new SensitivityAnalysisInput(network, context, actionsService, filterService);
-        sensitivityAnalysisInput.build(reporter);
+        SensitivityAnalysisInputBuilder sensitivityAnalysisInputBuilder = new SensitivityAnalysisInputBuilder(network, context, actionsService, filterService);
+        sensitivityAnalysisInputBuilder.build(reporter);
 
         CompletableFuture<SensitivityAnalysisResult> future = runSensitivityAnalysisAsync(context, sensitivityAnalysisRunner,
             network,
-            sensitivityAnalysisInput.getFactors(),
-            sensitivityAnalysisInput.getContingencies(),
-            sensitivityAnalysisInput.getVariableSets(),
+            sensitivityAnalysisInputBuilder.getFactors(),
```

Comment:
Why we do not construct the `SensitivityAnalysisInputBuilder` in the `runSensitivityAnalysisAsync` method